velox/common/memory/ByteStream.h

```diff
@@ -102,6 +104,11 @@ class ByteInputStream {
     current_ = &ranges_[0];
   }
 
+  /// Constructs 'this' to range over 'stream's current content. Does
+  /// not copy buffers. 'this' remains valid as long as 'stream' is
+  /// live and unchanged.
+  explicit ByteInputStream(const ByteStream& stream);
```

Comment:
Why do we need this? Looks like we are brining back confusion and mixing of input and output streams.